pkg/rancher-desktop/pages/images/add.vue

```diff
@@ -55,6 +55,7 @@ export default {
       imageToPull:                      '',
       imageOutputCuller:                null,
       showOutput:                       false,
+      allowedImagesEnabled:             null,
```

Comment:
This should be initialized to `false` because it comes from a boolean value. I don't think we need to later distinguish an unset value from `false` (although if we do, there should be a comment explaining why this needs to be tri-state (`true`, `false`, or `null`).

Reply:
`allowedImagesEnabled: null` is used in combination with `settings-update` messages to determine if the allowList is enabled:
```
const enabled = this.allowedImagesEnabled ?? this.preferences.containerEngine.imageAllowList.enabled;
```
- When the component is rendered, enabled variable is equal to `this.preferences.containerEngine.imageAllowList.enabled`
- When the settings are updated by the user, the `allowedImagesEnabled` is used because it will be always `true` or `false`

So, I am using the store to get the first value and the `settings-update` events to update the variable when the preferences changes.
Agree with you, even if we already have the first value in memory, this could be a bit confusing by code perspective, so using `once('settings-read')` to determine the first value would work fine.